src/components/userSignup/Signup.jsx

```diff
@@ -0,0 +1,126 @@
+import React, { useState } from "react";
+import { useNavigate } from "react-router-dom";
+import "./Signup.css";
+
+const Signup = () => {
+	const [email, setEmail] = useState("");
+	const [password, setPassword] = useState("");
+	const [name, setName] = useState("");
+	const [error, setError] = useState("");
+	const [emailError, setEmailError] = useState("");
+	const [showPassword, setShowPassword] = useState(false);
+	const [loading, setLoading] = useState(false);
+	const navigate = useNavigate();
+
+	const validateEmail = (email) => /^[^\s@]+@[^\s@]+\.[^\s@]+$/.test(email);
+
+	const validatePassword = (password) => {
+		const passwordRegex =
+			/^(?=.*[A-Za-z])(?=.*\d)(?=.*[@$!%*?&])[A-Za-z\d@$!%*?&]{6,}$/;
+		return passwordRegex.test(password);
+	};
+
+	const handleSubmit = async (e) => {
+		e.preventDefault();
+		setError("");
+		setEmailError("");
+
+		if (!name || !email || !password) {
+			setError("All fields are required.");
+			return;
+		}
+		if (!validateEmail(email)) {
+			setEmailError("Invalid email format.");
+			return;
+		}
+		if (!validatePassword(password)) {
+			setError(
+				"Password must include at least one letter, one number, and one special character, and be at least 6 characters long.",
+			);
+			return;
+		}
+
+		setLoading(true);
+
+		try {
+			const response = await fetch(
+				`${import.meta.env.VITE_API_BASE_URL}/api/v1/auth/register`,
+				{
+					method: "POST",
+					headers: { "Content-Type": "application/json" },
+					body: JSON.stringify({ name, email, password }),
+				},
+			);
+
+			if (response.ok) {
+				const data = await response.json();
+				localStorage.setItem("token", data.token);
```

Comment:
Just highlighting that we should update this to use our AuthContext from the login MR before merging once that has been merged in.